docs/topics/generics.md

```diff
@@ -322,6 +322,73 @@ fun <T> copyWhenGreater(list: List<T>, threshold: T): List<String>
 The passed type must satisfy all conditions of the `where` clause simultaneously. In the above example, the `T` type
 must implement _both_ `CharSequence` and `Comparable`.
 
+## Definitely non-nullable types
+
+To make interoperability with generic Java classes and interfaces easier, Kotlin supports declaring a generic type parameter
+as **definitely non-nullable**. 
+
+To declare a generic type `T` as definitely non-nullable, declare the type with `& Any`. For example: `T & Any`. 
+
+> The syntax represents an intersection between type `T` and the non-nullable type `Any`.
+> 
+{type="note"}
+
+A definitely non-nullable type must have a nullable [upper bound](#upper-bounds). In addition, `Any` must resolve to `kotlin.Any`.
+
+The most common use case for declaring definitely non-nullable types is when you want to override a Java method that 
+contains `@NotNull` as an argument. For example, consider the `bar()` method:
+
+```java
+import org.jetbrains.annotations.*;
+
+public interface A<T> {
+    public T foo(T x) { return x; }
+    @NotNull
+    public T bar(@NotNull T x) {}
+}
+```
+
+To override the `bar()` method in Kotlin successfully, you need `T` (and its subtypes) to be declared as definitely non-nullable:
+
+```kotlin
+interface B<T1> : A<T1> {
+    override fun foo(x: T1): T1
+    //Subtype T1 is definitely non-nullable
```

Comment:
I'm not sure 'subtype' is a correct term here, as it is just a generic substitution T1 -> T

Reply:
Fixed!